### DISCUSSION TITLE
Truncated BAM file causing samtools error

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1319] [Q&A] [answered]
@sr320 I'm using the deduplicated and sorted BAM files you generated for the Cv gonad samples (saved [here](https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/071521-cvBS/)). I need to merge all the files before calling SNPs. Tried doing so in [this Jupyter notebook](https://github.com/epigeneticstoocean/2018_L18-adult-methylation/blob/main/code/04-BS-Snper.ipynb) but got the following error:

![Screen Shot 2021-10-20 at 1 55 34 PM](https://user-images.githubusercontent.com/22335838/138147142-f70b099b-a0b0-4cb7-8996-cd7f9c132093.png)

Is there a reason why I'm getting this error/how would I check if this file is actually truncated? File sizes for reference:
![Screen Shot 2021-10-20 at 2 04 35 PM](https://user-images.githubusercontent.com/22335838/138173733-e05e657b-9937-481f-9262-2d7854aa0ec7.png)

edited to include the correct second screenshot and notebook link.

**Comment by kubu4** [comment]
A few things:

1. How did you transfer the files to your computer (i.e. via `rsync`, `wget`, `curl`, etc)?
2. If you did _not_ use `rsync`, did you verify checksums after transfer? Also, just re-download using `rsync` instead - no need to verify checksums.
3. Does your computer have sufficient space (both HDD and memory)? That's a LOT of BAM files...
4. Your link to your Jupyter notebook actually just links to this GitHub Discussion.

**Reply by yaaminiv** [reply]
1. Did not transfer the files at all. I'm working directly from the `gannet` folder (mounted it onto my computer)
3. I'm writing directly to `gannet`, so there should be enough memory!
4. Whoops. Notebook: https://github.com/epigeneticstoocean/2018_L18-adult-methylation/blob/main/code/04-BS-Snper.ipynb

**Comment by kubu4** [comment, marked as answer]
Ah! You will not (and should not) be able to work this way. The files are too large to transfer over the internet while processing them. Also, processing files this way is MASSIVELY slower than running things locally, due to the data transfer requirements back and forth, combined with the slower processor of Gannet having to handle that stuff simultaneously.

**Reply by kubu4** [reply]
In regards to Number 2, memory (RAM) and hard drive (HDD) are different things. The memory to run this is still based off of what amount of RAM your computer has (since this is what's running the commands).
